prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -317,27 +317,68 @@ int64_t rxr_pkt_req_cq_data(struct rxr_pkt_entry *pkt_entry)
 	return cq_data_hdr->cq_data;
 }
 
-size_t rxr_pkt_req_max_header_size(int pkt_type)
+
+size_t rxr_pkt_calc_data_size(struct rxr_ep *ep,
+		              fi_addr_t addr,
+			      int pkt_type,
+			      uint64_t fi_flags,
+			      size_t rma_iov_count)
 {
-	/* max_hdr_size does not include optional connid hdr length because
-	 * it is impossible to have both optional connid hdr and opt_raw_addr_hdr
-	 * in the header, and length of opt raw addr hdr is larger than
-	 * connid hdr (which is confirmed by the following assertion).
-	 */
-	assert(RXR_REQ_OPT_RAW_ADDR_HDR_SIZE >= sizeof(struct rxr_req_opt_connid_hdr));
+	struct rdm_peer *peer;
+	uint16_t header_flags = 0;
+
+	peer = rxr_ep_get_peer(ep, addr);
+	assert(peer);
+
+	if (peer->is_local) {
+		assert(ep->use_shm);
+		return rxr_env.shm_max_medium_size;
+	}
 
-	int max_hdr_size = REQ_INF_LIST[pkt_type].base_hdr_size
-		+ RXR_REQ_OPT_RAW_ADDR_HDR_SIZE
-		+ sizeof(struct rxr_req_opt_cq_data_hdr);
+	if (rxr_peer_need_raw_addr_hdr(peer))
+		header_flags |= RXR_REQ_OPT_RAW_ADDR_HDR;
+	else if (rxr_peer_need_connid(peer))
+		header_flags |= RXR_PKT_CONNID_HDR;
+
+	if (fi_flags & FI_REMOTE_CQ_DATA)
+		header_flags |= RXR_REQ_OPT_CQ_DATA_HDR;
+
+	return ep->mtu_size - rxr_pkt_req_max_header_size_extended(pkt_type,
+								header_flags,
+								rma_iov_count);
+}
+
+size_t rxr_pkt_req_max_header_size_extended(int pkt_type, uint16_t flags, size_t rma_iov_count)
```

Comment:
the function name should be `rxr_pkt_req_header_size` (drop `max`) because it calculate the  header size precisely.

Reply:
Also for both functions, please add a doxygen style function header.